src/pages/ProductDetail/ImgArea/ImgArea.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from 'react';
+
+export default function ImgARea(props) {
+  const [isImgStatus, setImgStatus] = useState(0);
+  const [mainImg, setMainImg] = useState();
+
+  useEffect(() => {
+    setMainImg(props.productData.images && props.productData.images[0].url);
+  }, [props]);
```

Comment:
props에 따라 컴포넌트가 리렌더링 되기 때문에, props를 의존성배열에 넣어두실 필요가 없습니다. 또한, props 데이터를 state로 저장하는 것 또한 권장되지 않는 방법입니다.
- [React로 사고하기 중](https://ko.reactjs.org/docs/thinking-in-react.html)
![image](https://user-images.githubusercontent.com/74031598/203563832-b5d4ae7f-ef1b-40d3-8927-82ab675c7119.png)

그러나 반드시 props를 state에 활용해야 한다면, useEffect를 사용할 필요 없이 해당 state의 초기값으로 넣으시면 될 것 같네요!

또한, isImgStatus라는 state명은 명확하지 않은 것 같습니다.
```suggestion
  const [imgIdx, setImgIdx] = useState(0);
  const [mainImg, setMainImg] = useState(props.productData.images?[0].url));
```

Reply:
그리고 상위 컴포넌트에서 productData가 없을 때의 처리를 해주었기 때문에, productData.images가 옵셔널하게 들어오는 데이터가 아니라면 조건 자체도 필요 없을 것 같습니다.
+props 구조분해할당 해주세요!
```suggestion
  const [imgIdx, setImgIdx] = useState(0);
  const [mainImg, setMainImg] = useState(props.productData.images[0].url));
```